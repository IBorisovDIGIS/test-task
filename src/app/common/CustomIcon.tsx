import { Icon, makeStyles } from '@material-ui/core';
import React from 'react';
import clsx from 'clsx';

const useStyles = makeStyles({
  imageIcon: {
    height: 24,
    width: 24,
  },
  iconRoot: {
    textAlign: 'center',
    height: '100%',
  },
});

interface ICustomIcon {
    icon: string;
    iconClasses?: string;
    alt?: string
}

const CustomIcon: React.FC<ICustomIcon> = ({ icon, iconClasses, ...props }: ICustomIcon) => {
  const classes = useStyles();

  return (
    <Icon classes={{ root: classes.iconRoot }}>
      <img
        className={clsx(classes.imageIcon, iconClasses)}
        src={icon}
        alt="alt"
        {...props}
      />
    </Icon>
  );
};

export default CustomIcon;

CustomIcon.defaultProps = {
  iconClasses: '',
  alt: '',
};
