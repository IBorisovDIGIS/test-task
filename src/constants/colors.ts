export const black = '#1B1D29';
export const white = '#fff';
export const gray = '#7D7D7D';
export const mainText = '#262626';
export const blue = '#0085FF';
