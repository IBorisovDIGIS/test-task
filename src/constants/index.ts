import menu from '../assets/icons/menu.svg';

export const menuItems = [
  {
    label: 'Dashboard',
    icon: menu,
  },
  {
    label: 'Services',
    icon: menu,
  },
  {
    label: 'Account',
    icon: menu,
  },
  {
    label: 'Messages',
    icon: menu,
  },
  {
    label: 'Test',
    icon: menu,
  },
];

export const languages = ['en', 'ru', 'fr'];

export const drawerWidth = 240;
