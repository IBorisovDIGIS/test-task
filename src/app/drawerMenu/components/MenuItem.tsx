import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { makeStyles } from '@material-ui/core';
import { white } from '../../../constants/colors';
import CustomIcon from '../../common/CustomIcon';

const useStyles = makeStyles({
  itemContainer: {
    padding: 24,
  },
  imageIcon: {
    height: 24,
    width: 24,
  },
  iconRoot: {
    textAlign: 'center',
    height: '100%',
  },
  text: {
    color: white,
  },
});

interface IMenuItem {
    icon: string;
    label: string;
}

const MenuItem: React.FC<IMenuItem> = ({ icon, label }: IMenuItem) => {
  const classes = useStyles();

  return (
    <ListItem button classes={{ root: classes.itemContainer }}>
      <ListItemIcon>
        <CustomIcon
          icon={icon}
          iconClasses={classes.imageIcon}
          alt={label}
        />
      </ListItemIcon>
      <ListItemText primary={label} classes={{ root: classes.text }} />
    </ListItem>
  );
};

export default MenuItem;
