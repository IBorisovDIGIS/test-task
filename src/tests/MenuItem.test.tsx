import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Container, Grid, ListItem, ListItemIcon, ListItemText, Typography} from '@material-ui/core';
import toJson from 'enzyme-to-json';
import MenuItem from '../app/drawerMenu/components/MenuItem';
import {menuItems} from "../constants";
import CustomIcon from "../app/common/CustomIcon";

const menuItem = menuItems[0];

configure({ adapter: new Adapter() });

describe('Menu item should render correctly', () => {
  it('Should renders correctly', () => {
    const container = shallow(<MenuItem label={menuItem.label} icon={menuItem.icon} />);

    expect(toJson(container)).toMatchSnapshot();
  });

  it('Should display icon correctly', () => {
    const item = shallow(<MenuItem label={menuItem.label} icon={menuItem.icon} />);
    expect(ListItem).toBeDefined();
    expect(ListItemIcon).toBeDefined();
    expect(CustomIcon).toBeDefined();

    const icon = item.find(CustomIcon).shallow().find('img');

    expect(icon.prop('src')).toEqual(menuItem.icon);
    expect(icon.prop('alt')).toEqual(menuItem.label);
  });

  it('Should display label correctly', () => {
    const item = shallow(<MenuItem label={menuItem.label} icon={menuItem.icon} />);
    expect(ListItem).toBeDefined();
    expect(ListItemText).toBeDefined();

    const text = item.find(ListItemText).prop('primary');
    expect(text).toEqual(menuItem.label);
  });

});
