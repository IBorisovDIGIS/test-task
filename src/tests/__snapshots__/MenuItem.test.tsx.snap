// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu item should render correctly Should renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  classes={
    Object {
      "root": "makeStyles-itemContainer-1",
    }
  }
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <CustomIcon
      alt="Dashboard"
      icon="menu.svg"
      iconClasses="makeStyles-imageIcon-2"
    />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    classes={
      Object {
        "root": "makeStyles-text-4",
      }
    }
    primary="Dashboard"
  />
</WithStyles(ForwardRef(ListItem))>
`;
